### DISCUSSION TITLE
Loading button in React with twin.macro

### DISCUSSION THREAD

**Discussion by fini** [saadeghi/daisyui#791] [Ideas / feature requests]
I have a big implementation of DaisyUI in React, and everything in general just works. I am using twin.macro to dynamically set the classes, and for some reason 'btn loading' simply doesn't work. I guess it's more related to twin.macro, but I was wondering if anyone have a solution/workaround to get it working dynamically. it seems the button does not get the before:: pseudo styling.
any assistance would be a great help! - thanks for DaisyUI - I'm loving it!

```
import tw, { styled, css } from 'twin.macro'
// ....
const StyledButton = styled.button(
  ({ variant, load, outline, wide, size, rounded }) => [
    tw`btn`,
    variantStyles[variant],
    outline && tw`btn-outline`,
    load && tw`loading`,
    wide && tw`btn-wide`,
    roundedStyles[rounded],
    sizeStyles[size],
  ],
)
```

**Comment by saadeghi** [comment]
I wish I could fix this on my end but daisyUI is basically just a set of CSS rules. If the result file has missing classes, it could be a bug from the CSS processor or It could be just incompatibility because of how the processor works.  
There are issues with Twin Macro. There are also issues with Windi CSS but I can't fix them on my end.  

Follow this discussion on Twin Macro repo please: https://github.com/ben-rogerson/twin.macro/discussions/549

(I'm moving this issue to discussions)
